src/components/AngerNoodle/helperFunctions/gameLogic.js

```diff
@@ -0,0 +1,110 @@
+import { makeRandomPositionIndex, copyBoard, copySnake, angryMessages } from "./data";
+
+const computerNextFrame = (shouldDeleteTail, updateShouldDeleteTail, snake, updateSnake, gridSize, board, updateBoard, direction, wallsAreLava, treatCoords, updateTreatCoords, currentScore, updateScore, highScore, updateHighScore, gameStateRef, snakeRef, updateAngryMessage) => {
```

Comment:
You have a tooooooon of items in here because this function does a lot. Instead of passing everything into this giant function that gets spit out into a different file I'd recommend breaking this all into several tiny function that each end up getting exported into your react component thats using it. In other words, this file should export all of the helper functions you're creating between lines 14 and 67 and lines 69-107 should be done in the main component (and maybe have further functions exported from here to make them tinier). By doing that, you'll turn this into a large interface of tiny functions that each take ~2-3 of the above variables that you can flexibly toss around where they're needed instead of having one function that takes ~20 variables.

Another option - if you want to continue using the oop class logic that I remember you mentioning that you liked you can turn this file into a "Game" class with these functions living on methods on them, even going a step further to create a custom hook called "useSnakeGame" that makes the snake specific methods more readily available to you